components/org.wso2.carbon.identity.user.endpoint/src/test/java/org.wso2.carbon.identity.user.endpoint.Util/UtilsTest.java

```diff
@@ -0,0 +1,165 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.user.endpoint.Util;
+
+import com.beust.jcommander.internal.Lists;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.user.endpoint.Exceptions.BadRequestException;
+import org.wso2.carbon.identity.user.endpoint.Exceptions.ConflictException;
+import org.wso2.carbon.identity.user.endpoint.Exceptions.InternalServerErrorException;
+import org.wso2.carbon.identity.user.endpoint.dto.ClaimDTO;
+import org.wso2.carbon.identity.user.endpoint.dto.ErrorDTO;
+import org.wso2.carbon.identity.user.endpoint.dto.PropertyDTO;
+import org.wso2.carbon.identity.user.endpoint.dto.SelfRegistrationUserDTO;
+import org.wso2.carbon.identity.user.endpoint.dto.UserDTO;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class contains unit tests required for UtilsTest.java.
+ */
+
+public class UtilsTest {
+
+    private static final String INTERNAL_SERVER_ERROR_CODE = "500";
+    private static final String BAD_REQUEST_ERROR_MESSAGE = "Bad Request";
+
+    @Test(expectedExceptions = {InternalServerErrorException.class})
+    public void testInternalServerErrorFromCode() {
+
+        throw Utils.buildInternalServerErrorException(INTERNAL_SERVER_ERROR_CODE);
+    }
+
+    @Test(expectedExceptions = {BadRequestException.class})
+    public void testHandleBadRequest() {
+
+        Utils.handleBadRequest(INTERNAL_SERVER_ERROR_CODE, BAD_REQUEST_ERROR_MESSAGE);
+    }
+
+    @Test(expectedExceptions = {BadRequestException.class})
+    public void testBuildBadRequestException() {
+
+        throw Utils.buildBadRequestException(INTERNAL_SERVER_ERROR_CODE, BAD_REQUEST_ERROR_MESSAGE);
+    }
+
+    @Test(expectedExceptions = {ConflictException.class})
+    public void testHandleConflict() {
+
+        Utils.handleConflict(INTERNAL_SERVER_ERROR_CODE, BAD_REQUEST_ERROR_MESSAGE);
+    }
+
+    @Test(expectedExceptions = {ConflictException.class})
+    public void testBuildConflict() {
+
+        Utils.handleConflict(INTERNAL_SERVER_ERROR_CODE, BAD_REQUEST_ERROR_MESSAGE);
+    }
+
+    @Test
+    public void testErrorDTO() {
+
+        ErrorDTO errorDTO = Utils.getErrorDTO(BAD_REQUEST_ERROR_MESSAGE, INTERNAL_SERVER_ERROR_CODE, BAD_REQUEST_ERROR_MESSAGE);
+        Assert.assertNotNull(errorDTO, "Failed building error message.");
+    }
+
+    @Test
+    public void testGetSelfRegisteredUser() {
+
+        Assert.assertNotNull(Utils.getUser(buildSelfRegistrationUserDTO()), "Failed building self registered user object.");
+    }
+
+    @Test
+    public void testGetUser() {
+
+        Assert.assertNotNull(Utils.getUser(buildUserDTO()), "Failed building user object.");
+        Assert.assertNotNull(Utils.getUser(buildUserDTO()).getUserName(), "Failed retrieving user name.");
```

Comment:
getUsername could still provide empty string or something, better to assert the exact username we expect